src/haddock/libs/libpdb.py

```diff
@@ -57,6 +58,18 @@ def split_by_chain(pdb_file_path):
     return get_new_models(pdb_file_path)
 
 
+def merge(pdb_file_list, new_pdb_file_path):
```

Comment:
Honestly, I would use `pdbtools` here:

```python
from pdbtools import pdb_merge, pdb_tidy

def merge(pdb_file_list, new_pdb_file_path):
    input_fhs = (open(p) for p in pdb_file_list)
    _ = pdb_merge.run(input_fhs)
    merged = pdb_tidy.run(_)
    Path(new_pdb_file_path).write_text(os.linesep.join(merged))
```

I think there is no typo bug, the function should work as is. Also, your function is not closing the input files for you. While `pdb_merge` does it :+1:

Reply:
this is `pdbtools`, check on the top `from pdbtools.pdb_merge import concatenate_files`

Reply:
Yes, but your code repeats the `concatenate_files`. You are sending one file at the time, and getting one line at the time. Also `pdbmerge` is quite simple, if you don't want to use it, it fine. But the present code is using and desusing it :wink: